core/src/checks/check_sign_tx.c

```diff
@@ -126,6 +133,62 @@ int verify_sign_tx(void) {
     print_bytes(resp.signatures[0].der.bytes, resp.signatures[0].der.size);
     return -1;
   }
+  
   INFO("sign_tx passed");
   return 0;
 }
+/**
+ * This function does not make an effort to zeroize
+ * anything as all of the values are test values.
+ * It will just test our verify signature wrapper.
+ */
+
+int verify_check_signature_pub(void){
+  //random ECDSA private key.
+  unsigned char private_key[32] = {
+    0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
+    0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
+    0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
+    0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01
+  };
+  unsigned char buf[8] = {
+    0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01
+  };
+  unsigned char signature[64] = {0};
+  if(0 != ecdsa_sign(
+      (const ecdsa_curve *)&nist256p1,
+      HASHER_SHA2,
+      private_key,
+      buf,
+      8,
+      signature,
+      NULL,
+      NULL
+    )
+  ){
+    ERROR("Could not sign during self check.");
+    return 1;
+  }
+  unsigned char pub[65] = {0};
+  ecdsa_get_public_key65((const ecdsa_curve *)&nist256p1, private_key, pub);
+  if (!check_signature_pub(buf, 8, signature, pub)) {
+    ERROR("verify signature does not seem to work.");
+    return 1;
```

Comment:
nit: we're returning `1` in two cases, maybe worth distinguishing those? no hard feelings since there's also the ERROR message.